### ISSUE TITLE
Separate Registry Support

### ISSUE THREAD

**Issue by nerumo** [droidsolutions/semantic-release-nuget#624]
It's a common use case that within company, you use one gitlab project as the registry for all the packages. Therefore, if I have a nuget project (lets say project 1) that uses a registry (project 2), then I have a conflict with the CI_PROJECT_ID variable.

Of course I could set the CI_PROJECT_ID variable to "2", but then the semantic-release/gitlab publish doesn't work anymore, since it tries to create releases with a tag, that does not exist on project 2. 

And since neither of the projects can have a separate variable to configure the project id, I'm in conflict. 

I'd need a separate `gitlabRegistryProjectId` to control where the nuget package should be pushed

**Comment by Kampfmoehre** [comment]
I am not sure if I understand fully, you want to publish a NuGet package of project 1 to the package registry of project 2? I did a quick read on [the docs](https://docs.gitlab.com/ee/user/packages/nuget_repository/#publish-a-nuget-package) and I am not sure if this is even allowed on GitLab.

If all you need to change is the project id, you could try to overwrite the `CI_PROJECT_ID` variable in your ci like this:

```yaml
release:
  variables:
    CI_PROJECT_ID: 2
  script:
    - npx semantic-release
```

This should have the same effect. 
If I did not understand your issue correctly, please explain a little more.

**Comment by Kampfmoehre** [comment]
@nerumo can you test version `1.3.0-beta.1` with your workflow? As far as I know, `CI_JOB_TOKEN` is only scoped to the current project, so the plugin uses `NUGET_TOKEN` when `gitlabRegistryProjectId` is set. Make sure you provide a GitLab token with access to the other GitLab project.

**Comment by Kampfmoehre** [comment]
@nerumo where you able to test the new version?

**Comment by nerumo** [comment]
thank you for your quick action. Sorry for the delay. I tested it now. It correctly replaces the publish project for gitlab :clap: . But now it does not consider the NUGET_TOKEN anymore. In the last version, this was working. Here's the output:

![image](https://github.com/droidsolutions/semantic-release-nuget/assets/8597645/fcfacbf9-48ff-4b67-8a7b-a9b960399c74)

I printed the NUGET_TOKEN before the publish and it contains a correct value

**Comment by nerumo** [comment]
Sorry, I did not read carefully enough. As you wrote, the behavior changed which api key is used. I created an access token for the registry project with all rights (just to make sure), used it as NUGET_TOKEN and the result stayed the same. I then used a personal access token (full access), the result was still the same. I couldn't find your changes to see where the problem could be

**Comment by Kampfmoehre** [comment]
I have some questions to further investigate the problem:
1. Can you confirm that the push Url correct, e.g. checking the project id is correct? 
2. Have you followed [the GitLab docs](https://docs.gitlab.com/ee/user/packages/nuget_repository/#add-the-package-registry-as-a-source-for-nuget-packages) to create a token with the correct rights?

I suspect we may also need to give the username along with the token to Semantic Release for this to work. Right now it is hardcoded `gitlab-ci-token` which works for CI_JOB_TOKEN but if you use deploy token this should probably be the name of the token or when using a PAT (like in your case) the name of the user may be needed here.

**Comment by nerumo** [comment]
yes, push url and project is correct. according to the gitlab documentation we have to pass the username. You're probably right with your assumption. So another parameter/env variable is needed. Tell me if I can be of any help

**Comment by Kampfmoehre** [comment]
@nerumo can you test if it works with 1.3.0-beta.2? You should be able to set `gitlabUser` in the config. Thinking about it, I should have probably also added the ability to set it via env var.

**Comment by nerumo** [comment]
Awesome, it worked. Thank you very much

**Comment by Kampfmoehre** [comment]
@nerumo is now released as [1.3.0](https://www.npmjs.com/package/@droidsolutions-oss/semantic-release-nuget/v/1.3.0).
